pgcli/packages/pgspecial/main.py

```diff
@@ -18,6 +18,32 @@
 class CommandNotFound(Exception):
     pass
 
+
+@export
+class PGSpecial(object):
+    def __init__(self):
+        self.timing_enabled = True
+
+        self.commands = COMMANDS.copy()
+        self.register(self.show_help, '\\?', '\\?', 'Show Help.',
+                      arg_type=NO_QUERY)
+
+    def register(self, *args, **kwargs):
+        register_special_command(*args, command_dict=self.commands, **kwargs)
```

Comment:
You're right. This does create a complication. :(

Here's a thought. 

We have two sets of special commands. 

1. Default Special Commands
2. Custom Special Commands

Default special commands come with the library and they are defined as functions in this package. Custom special commands are registered using the hook. 

Default special commands can be saved under a class variable in PGSpecial class. The Custom special commands are stored in an instance variable in the PGSpecial object.

We have one decorator which is defined as a class method on PGSpecial. This decorator will only be used by the library author to add default special commands to the class var. So we can keep the `describe_table`, `list_views` etc as unbound functions and decorate them using this decorator. This is merely for convenience at this point. 

Custom special commands will have to be added using the reigster_special_command() method which can only be called on an instance of the PGSpecial class. 

So we eschew the `@special_decorator` for external use and only use it for internal purposes and change it to add commands to the class var. 

The absence of decorator for external use is fine in this case, because users can still define their custom commands using the register_special_command() method. 

This idea is fairly new and I haven't thought this through completely, so feel free to disagree.